airflow/models/backfill.py

```diff
@@ -288,6 +289,8 @@ def _create_backfill_dag_run(
 ):
     from airflow.models.dagrun import DagRun
 
+    # clear dag run if run exits and reprocess behaviour is in completed or failed
+    clear_run_if_dagrun_exists(dag, info, reprocess_behavior, backfill_id, session)
```

Comment:
i feel like `clear_run_if_dagrun_exists` should be moved into `should_create_backfill_dag_run`.  we should probably only evaluate the reprocess behavior in _one place_.  and it should only be evaluated against runs that are there before the backfill request, i.e. not those modified (i.e. cleared) in the backfill request.